crates/orchestrator/src/queue/job_queue.rs

```diff
@@ -198,8 +198,8 @@ fn get_worker_handler_from_worker_trigger_type(worker_trigger_type: WorkerTrigge
 }
 
 /// To get the delivery from the message queue using the queue name
-async fn get_delivery_from_queue(queue: String) -> Result<DeliveryReturnType, ConsumptionError> {
-    match config().await.queue().consume_message_from_queue(queue.clone()).await {
+async fn get_delivery_from_queue(queue: &str) -> Result<DeliveryReturnType, ConsumptionError> {
+    match config().await.queue().consume_message_from_queue(queue.to_string()).await {
```

Comment:
the `.to_string()` actually beats the purpose because it's still being cloned. can we actually optimise the `consume_message_from_queue` method to work on &str or a reference instead?